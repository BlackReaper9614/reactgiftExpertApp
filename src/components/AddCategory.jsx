import { useState } from 'react';

export const AddCategory = ( { onNewCategory } ) => {
  
    const [inputValue, setInputValue] = useState('');

    const onInutChange = (event) => {
        
        setInputValue(event.target.value);

    }

    const onSubmit = (event) => {
        
        event.preventDefault();

        if(inputValue.trim().length < 1)
        {
            return;
        }

        onNewCategory( inputValue.trim() );

        setInputValue('');
        
    }

    return (
        <>
            <form onSubmit={ onSubmit }>

                <input
                    id="inputCategory" 
                    onChange={ onInutChange }
                    placeholder="Bucar gifs" 
                    type="text"  
                    value={ inputValue}
                />

            </form>

        </>

    )

}
