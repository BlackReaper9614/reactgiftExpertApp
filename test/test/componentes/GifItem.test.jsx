import { render, screen } from "@testing-library/react";
import { GifItem } from "../../../src/components/GifItem";

describe('Prueba de <GifItem>', () => { 

    const title = 'Pokemon';
    const url = 'https://holamundo.com/';

    test('Debe hacer match con el snapshot', () => { 
      
        const { container } = render( <GifItem  title={ title } url={ url }/> );

        expect( container ).toMatchSnapshot();

    })

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => { 
    
        render( <GifItem  title={ title } url={ url }/> );

        //screen.debug();

        const { src, alt } = screen.getByRole('img');

        expect( src ).toBe( url );

        expect( alt ).toBe( title );

    })

    test('Debe de mostrar el titulo que se envÃ­a', () => { 
    
        render( <GifItem  title={ title } url={ url }/> );

        expect( screen.getByText( title ) ).toBeTruthy();

    })

})