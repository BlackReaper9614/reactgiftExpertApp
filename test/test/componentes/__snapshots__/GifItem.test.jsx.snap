// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de <GifItem> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Pokemon"
      src="https://holamundo.com/"
    />
    <p>
      Pokemon
    </p>
  </div>
</div>
`;
